scipy/integrate/_quadrature.py

```diff
@@ -1029,20 +1033,19 @@ def cumulative_simpson(y, *, x=None, dx=1.0, axis=-1, initial=None):
     y = _ensure_float_array(y)
 
     # validate `axis` and standardize to work along the last axis
+    original_y = y.copy()
```

Comment:
```suggestion
    original_y = y
```
I don't think we need a complete copy of the data because `swapaxes` below doesn't modify it.